blocks/gallery-block/features/gallery/default.jsx

```diff
@@ -58,30 +71,86 @@ export const GalleryPresentation = ({
 
 	const interstitialClicks = parseInt(galleryCubeClicks, 10);
 
+	const galleryLength = contentElements.length;
+
 	return (
-		<Gallery
-			galleryElements={contentElements}
-			resizerURL={resizerURL}
-			ansId={id}
-			ansHeadline={headlines?.basic ? headlines.basic : ""}
-			expandPhrase={phrases.t("global.gallery-expand-button")}
+		<Carousel
+			className={BLOCK_CLASS_NAME}
+			showLabel
+			label={headlines?.basic ? headlines.basic : ""}
+			slidesToShow={1}
+			showAdditionalSlideControls
+			pageCountPhrase={
+				/* istanbul ignore next */ (current, total) =>
+					phrases.t("global.gallery-page-count-text", { current, total })
+			}
+			enableAutoplay
+			startAutoplayIcon={<Icon name="Play" />}
+			startAutoplayText={phrases.t("global.gallery-autoplay-button")}
+			stopAutoplayIcon={<Icon name="Pause" />}
+			stopAutoplayText={phrases.t("global.gallery-pause-autoplay-button")}
 			autoplayPhraseLabels={{
 				start: phrases.t("global.gallery-autoplay-label-start"),
 				stop: phrases.t("global.gallery-autoplay-label-stop"),
 			}}
-			autoplayPhrase={phrases.t("global.gallery-autoplay-button")}
-			pausePhrase={phrases.t("global.gallery-pause-autoplay-button")}
-			controlsFont={getThemeStyle(arcSite)["primary-font-family"]}
-			pageCountPhrase={
-				/* istanbul ignore next */ (current, total) =>
-					phrases.t("global.gallery-page-count-text", { current, total })
+			enableFullScreen
+			fullScreenShowButton={
+				<button type="button">
+					<Icon name="Fullscreen" />
+					{phrases.t("global.gallery-expand-button")}
+				</button>
+			}
+			fullScreenMinimizeButton={
+				<button type="button">
+					<Icon name="Close" />
+				</button>
 			}
 			adElement={/* istanbul ignore next */ () => <AdBlock />}
```

Comment:
The react warning you are seeing for `adElement` is because you are passing a function as the prop this should be 
```suggestion
			adElement={/* istanbul ignore next */ <AdBlock />}
```

Reply:
updated